File: exercism/dart/hamming/lib/hamming.dart
class Hamming {
  int distance(String leftStrand, String rightStrand) {
    var hammingDistance = 0;
    if (leftStrand.length != rightStrand.length) {
      throw new ArgumentError("strands must be of equal length");
    }

    for (var i = 0; i < leftStrand.length; i++) {
      if (leftStrand.substring(i, i + 1) != rightStrand.substring(i, i + 1)) {
        hammingDistance++;
      }
    }
    return hammingDistance;
  }
}

File: exercism/kotlin/hamming/src/main/kotlin/Hamming.kt
object Hamming {

    fun compute(leftStrand: String, rightStrand: String): Int {
        var hammingDistance = 0
        if (leftStrand.length != rightStrand.length) {
            throw IllegalArgumentException("left and right strands must be of equal length")
        }

        for (i in leftStrand.indices) {
            if (leftStrand.get(i) !== rightStrand.get(i)) {
                hammingDistance++
            }
        }
        return hammingDistance
    }
}
